wrappers/python/examples/roberta_training.py

```diff
@@ -0,0 +1,313 @@
+x_distr = [0] * x_traits.grid.nelems
+
+rng = np.random.default_rng()
+for epoch_idx in range(args.nepochs):
+    masked_data_epoch = []
+    labels_epoch = []
+    for i in range(num_train_batches):
+        minibatch_masked_data = []
+        minibatch_labels = []
+        for j in range(num_minibatch):
+            current_mask = np.zeros((args.minibatch_size,
+                                     args.seq_len), dtype=bool)
+            idx_masked_tokens = rng.choice(args.seq_len,
+                                           size=(args.minibatch_size,
+                                                 args.n_masked_tokens_per_seq))
+            current_mask[:, idx_masked_tokens] = 1
+            x = nntile.tensor.Tensor_int64(x_traits, x_distr, next_tag)
+            next_tag = x.next_tag
+            current_minibatch = train_tokens[i, j, :, :-1].copy()
+            current_minibatch[current_mask] = args.label_mask_token
+            x.from_array(np.asfortranarray(current_minibatch).T)
+            minibatch_masked_data.append(x)
+            y = nntile.tensor.Tensor_int64(x_traits, x_distr, next_tag)
+            next_tag = y.next_tag
+            current_label = train_tokens[i, j, :, :-1].copy()
+            inverse_current_mask = np.array(1 - current_mask, dtype=bool)
+            # Ignore index -100
+            current_label[inverse_current_mask] = -100
+            y.from_array(np.asfortranarray(current_label.T))
+            minibatch_labels.append(y)
+        masked_data_epoch.append(minibatch_masked_data)
+        labels_epoch.append(minibatch_labels)
+    batch_masked_data.append(masked_data_epoch)
+    batch_labels.append(labels_epoch)
+time1 = time.time() - time0
+print("From PyTorch loader to NNTile batches in {} seconds".format(time1))
+# Set up learning rate and optimizer for training
+if args.optimizer == "adam":
+    optimizer = nntile.optimizer.Adam(bert_nntile.get_parameters(),
+            args.lr, next_tag)
+elif args.optimizer == "adamw":
+    optimizer = nntile.optimizer.AdamW(bert_nntile.get_parameters(),
+            args.lr, next_tag)
+elif args.optimizer == "sgd":
+    optimizer = nntile.optimizer.SGD(bert_nntile.get_parameters(),
+            args.lr, next_tag)
+next_tag = optimizer.get_next_tag()
+# Define Cross Entropy loss function
+loss, next_tag = nntile.loss.CrossEntropy.generate_simple(
+        bert_nntile.activations[-1], next_tag,
+        scale=1.0 / (args.batch_size *
+                     args.n_masked_tokens_per_seq))
+# Set up training pipeline
+pipeline = nntile.pipeline.Pipeline(batch_masked_data[0], batch_labels[0],
+        bert_nntile, optimizer, loss, 1)
+
+# Print pipeline memory info
+pipeline.print_meminfo()
+# Warmup training
+# nntile.starpu.pause()
+nntile.starpu.profiling_enable()
+# Call separate pipeline for the single epoch for every masked data
+for i in range(args.nepochs):
+    pipeline = nntile.pipeline.Pipeline(batch_masked_data[i], batch_labels[i],
```

Comment:
you create a new pipeline every epoch. Old pipeline might be deleted even before execution finishes. It may lead to very bad problems. We discussed making `batch_masked_data` contain inputs for all epochs without adding a list outside. It shall be [data1_epoch1, data2_epoch1, data1_epoch2, data2_epoch2]. What is done here is the following: [[data1_epoch1, data2_epoch1], [data1_epoch2, data2_epoch2]].